subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -849,4 +852,109 @@ $append
 
     }
 
+    def 'can publish java library using a platform with a dependencyManagement entry of type "pom" and scope "import"'() {
+        given:
+        javaLibrary(mavenRepo.module("org.test", "bom", "1.0")).hasPackaging('pom').dependencyConstraint(mavenRepo.module('org.test', 'bar', '1.1')).withModuleMetadata().publish()
+        javaLibrary(mavenRepo.module("org.test", "bar", "1.0")).withModuleMetadata().publish()
+        javaLibrary(mavenRepo.module("org.test", "bar", "1.1")).withModuleMetadata().publish()
+
+        createBuildScripts("""
+            dependencies {
+                api "org.test:bar"
+                api platform("org.test:bom:1.0")
```

Comment:
Should probably test publishing of a platform dependency on `implementation` as well:
- Will show up in the `runtime` variant (but not `api`) in Gradle module metadata
- Should be included in the POM with scope `import` (lossy mapping)